pkg/asset/machines/worker.go

```diff
@@ -108,6 +109,14 @@ func defaultOvirtMachinePoolPlatform() ovirttypes.MachinePool {
 	return ovirttypes.MachinePool{}
 }
 
+func defaultVSphereMachinePoolPlatform() vspheretypes.MachinePool {
+	return vspheretypes.MachinePool{
```

Comment:
I think these should be defaulted as well:
https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go#L34-L49